Source/control.cpp

```diff
@@ -264,16 +265,19 @@ void PrintInfo(const Surface &out)
 	const int LineStart[] = { 70, 58, 52, 48, 46 };
 	const int LineHeights[] = { 30, 24, 18, 15, 12 };
 
-	Rectangle line { GetMainPanel().position + Displacement { 177, LineStart[pnumlines] }, { 288, 12 } };
+	auto infoStringLines = (int)std::count(InfoString.str().begin(), InfoString.str().end(), '\n');
+
+	Rectangle line { GetMainPanel().position + Displacement { 177, LineStart[infoStringLines + pnumlines] }, { 288, 12 } };

```

Comment:
I mentioned this in another PR, but the array access here is not bounds checked. `AddPanelString()` is applying validation to the input to ensure that `pnumlines` doesn't cause an overflow, and this change subverts it. To do this right, you'd probably need to combine all the strings using newlines and then truncate the result if it's more than 5 lines. Supporting both `InfoString` and `AddPanelString()` together definitely complicates this.

Reply:
Took a stab at handling this